frontend revamp

IronJam11 opened the pull request:
### Summary

This PR introduces several key enhancements aimed at improving security, user experience, and code maintainability:

1. **Wallet Integration**

   * Implemented WalletConnect to replace the previously hardcoded wallet private key, enhancing security and flexibility.
   * Enabled deep linking support for popular wallets like MetaMask, RainbowKit, etc.

2. **Theme Management**

   * Introduced a `ThemeProvider` to handle light/dark mode and allow consistent theming across the app.

3. **Navigation System**

   * Integrated `go_router` for structured and declarative navigation throughout the application.

Connecting wallet through deeplinking using metamask : https://drive.google.com/file/d/1EO-Q87HWRz0DHPwyoLaoLlLfVKe8yMHf/view?usp=sharing 

Demo screenshots: 
<img width="460" alt="Screenshot 2025-07-01 at 2 50 56 PM" src="https://github.com/user-attachments/assets/ce4b196f-7949-4510-8cf6-7c36d44e32e9" />
<img width="480" alt="Screenshot 2025-07-01 at 2 51 03 PM" src="https://github.com/user-attachments/assets/31368d02-212a-4b3a-85ae-0dc5527a6694" />
<img width="593" alt="Screenshot 2025-07-01 at 2 51 10 PM" src="https://github.com/user-attachments/assets/e6edb1c0-a756-4d81-aa5a-afc6ad415d2a" />
<img width="490" alt="Screenshot 2025-07-01 at 2 51 17 PM" src="https://github.com/user-attachments/assets/378d5c61-0e0d-4001-a8e8-f51e98d03a02" />



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Introduced a new cross-platform Flutter app called "Tree Planting Protocol" with wallet connectivity, blockchain network support, and theme switching.
  * Added bottom navigation and universal app bar with wallet and chain management.
  * Implemented wallet connection dialog and blockchain network switching.
  * Provided utility functions for Ethereum value conversions and address formatting.

* **Bug Fixes**
  * Updated configuration files and manifests for improved platform compatibility and build stability.

* **Chores**
  * Added comprehensive `.gitignore` files for all platforms.
  * Updated project and app naming conventions.
  * Cleaned up and removed obsolete frontend, configuration, and asset files.
  * Added platform-specific build scripts and configuration files for Linux, macOS, and Windows.

* **Documentation**
  * Added and updated manifest and configuration files for web and desktop platforms.

* **Tests**
  * Added a placeholder unit test for the macOS application.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

bhavik-mangla commented on lib/providers/wallet_provider.dart:
```diff
@@ -0,0 +1,329 @@
+
+      _web3App = Web3App(
+        core: Core(
+          projectId: 'bae2d46d706b0bfd76087f472ffec96d',
```
@IronJam11 consider this isssue

bhavik-mangla commented on lib/pages/home_page.dart:
Initialize a GoRouter inside MyApp class like this instead. That would ensure smoother and faster screen loading, and maintaining previous screen contexts

Example - 
class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {

    final rootNavigatorKey = GlobalKey<NavigatorState>();
    final shellNavigatorHomeKey = GlobalKey<NavigatorState>(debugLabel: 'shellHome');
    final shellNavigatorTrendingKey = GlobalKey<NavigatorState>(debugLabel: 'shellTrees');
    final shellNavigatorProfileKey = GlobalKey<NavigatorState>(debugLabel: 'shellProfile');
    
    //etc.


    GoRouter router = GoRouter(


      initialLocation: '/',
      navigatorKey: rootNavigatorKey,
        overridePlatformDefaultLocation: true,

      routes: [


                    StatefulShellRoute.indexedStack(
                    builder: (context, state, navigationShell) {
                      // the UI shell
                      return ScaffoldWithNestedNavigation(
                          navigationShell: navigationShell);
                    },

                    branches: [
                      StatefulShellBranch(
                        navigatorKey: shellNavigatorHomeKey,
                        routes: [
                          // top route inside branch
                          GoRoute(
                            path: '/',
                            pageBuilder: (context, state) =>
                                NoTransitionPage(
                                  child: HomePageScreen(user: user),
                                ),

IronJam11 commented:
- Added a struct with four options for initialising 
   1. notStarted
   2. initializing
   3. initialized
   4. failed
- Added flutter_dotenv for storing the walletconnect projectID 
- Initialized a GoRouter inside MyApp
